app/controllers/redirection_controller.rb

```diff
@@ -1,28 +1,53 @@
 class RedirectionController < ApplicationController
   def announcements
-    parameters = { keywords: params['keywords'],
-                  level_one_taxon: params['taxons'].try(:first),
-                  level_two_taxon: params['subtaxons'].try(:first),
-                  people: params['people'],
-                  organisations: params['departments'],
-                  world_locations: params['world_locations'],
-                  public_timestamp: { from: params['from_date'], to: params['to_date'] }.compact.presence }
+    parameters = convert_common_parameters.merge(
+      level_two_taxon: params['subtaxons'].try(:first),
+      people: params['people'],
+      world_locations: params['world_locations']
+    )
     respond_to do |format|
       format.html { redirect_to(finder_path('search/news-and-communications', params: parameters)) }
       format.atom { redirect_to(finder_path('search/news-and-communications', params: parameters, format: :atom)) }
     end
   end
 
   def publications
-    parameters = { keywords: params['keywords'],
-                  level_one_taxon: params['taxons'].try(:first),
-                  level_two_taxon: params['subtaxons'].try(:first),
-                  organisations: params['departments'],
-                  world_locations: params['world_locations'],
-                  public_timestamp: { from: params['from_date'], to: params['to_date'] }.compact.presence }
+    parameters = convert_common_parameters.merge(
+      level_two_taxon: params['subtaxons'].try(:first),
+      world_locations: params['world_locations']
```

Comment:
I would recommend moving `people`, `level_two_taxon` and `world_locations` into the `convert_common_parameters` method.

If param keys like `world_locations` are unlikely to be used more in different ways, even if they are not used by all finders there's no harm in putting `world_locations` into the translated common params method at the bottom. You can always call `.compact` on the hash to remove keys with no values.

Reply:
Hey brill, I didn't know about .compact :D

Reply:
Alrighty, refactored so it's super condensed now